### PR TITLE
Add UI Text Entry Box Reference

### PR DESCRIPTION
**Pull request by CommanderFoo**
# Description

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by CommanderFoo** [comment]
TODO: Correct name in Create menu (Text Entry vs Editable Text, waiting on QA)
TODO: Add more Learn more links (i.e, tutorial)

File: src/references/ui_text_entry.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+| `Active Color` | The color when the user is editing the control. |
+| `Disabled Color` | The color when the control is disabled. |
+| `Selection Color` | The color of the text selection when highlighting all or part of the string. |
+| `Only Use Background Color` | If true, ignores Hovered, Active, and Disabled colors. |
+
+#### Behavior
+
+At runtime, a UI text entry object that **is interactable** can be clicked on with a mouse.
+
+When not being interacted with, the UI text entry object is in its background state. If the UI text entry object is empty and **Prompt** is true then the default text (using Prompt Color) is displayed. If the user has entered text or the creator has applied non-prompt text, the text is also displayed.
```
Comment (DoubleA-MG) [inline src/references/ui_text_entry.md]: the last sentence: "If the user has entered text or the creator has applied non-prompt text, then that text is displayed."

File: src/references/ui_text_entry.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+| `Is Interactable` | Whether text control is enabled. |
+| `Is Multiline` | The text field can span multiple lines. |
+| `Image` | The background image of the control. |
+| `Background Color` | The base color of the control. |
+| `Hovered Color` | The color when the cursor moves over the control. |
+| `Active Color` | The color when the user is editing the control. |
+| `Disabled Color` | The color when the control is disabled. |
+| `Selection Color` | The color of the text selection when highlighting all or part of the string. |
+| `Only Use Background Color` | If true, ignores Hovered, Active, and Disabled colors. |
+
```
Comment (DoubleA-MG) [inline src/references/ui_text_entry.md]: should `promptText` be added to the table?

File: src/references/ui_text_entry.md (hunk 1 above)
Comment (DoubleA-MG) [reply src/references/ui_text_entry.md]: Haven't used it yet, but is there a boolean property `Prompt` or is it just checking if `promptText` is not empty?

**Review by DoubleA-MG — CHANGES_REQUESTED** [review]

File: src/references/ui_text_entry.md (hunk 1 above)
Comment (CommanderFoo) [reply src/references/ui_text_entry.md]: It checks if the prompt text has a value. It can be left empty and nothing will show.

**Review by DoubleA-MG — APPROVED** [review]

**Pull request merged**